source/DomainTheory/Part-I.lagda

```diff
@@ -0,0 +1,819 @@
+        (fe : Fun-Ext)
+        (pe : Prop-Ext)
+        (pt : propositional-truncations-exist)
+       where
+
+open PropositionalTruncation pt
+
+open import MLTT.Spartan
+
+open import Naturals.Order hiding (subtraction')
+open import Naturals.Addition renaming (_+_ to _+'_)
+open import Notation.Order hiding (_⊑_ ; _≼_)
+
+open import UF.Base
+open import UF.Equiv
+open import UF.Hedberg
+open import UF.ImageAndSurjection pt
+open import UF.Powerset-MultiUniverse
+open import UF.Sets
+open import UF.Size hiding (is-locally-small)
+open import UF.Subsingletons-FunExt
+open import UF.SubtypeClassifier hiding (⊥)
+open import UF.Univalence
+
+open import OrderedTypes.Poset fe
+
+{- Section 2 -}
+
+Definition-2-1 : (𝓤 : Universe) (X : 𝓥 ̇  ) → 𝓤 ⁺ ⊔ 𝓥 ̇
+Definition-2-1 𝓤 X = X is 𝓤 small
+
+Definition-2-2 : (𝓤 : Universe) → 𝓤 ⁺ ̇
+Definition-2-2 𝓤 = Ω 𝓤
+
+Definition-2-3 : (𝓥 : Universe) (X : 𝓤 ̇  ) → 𝓥 ⁺ ⊔ 𝓤 ̇
+Definition-2-3 𝓥 X = 𝓟 {𝓥} X
+
+Definition-2-4 : (𝓥 : Universe) (X : 𝓤 ̇  )
+               → (X → 𝓟 {𝓥} X → 𝓥 ̇  )
+               × (𝓟 {𝓥} X → 𝓟 {𝓥} X → 𝓥 ⊔ 𝓤 ̇  )
+Definition-2-4 𝓥 X = _∈_ , _⊆_
+
+{- Section 3 -}
+
+module _
+        (P : 𝓤 ̇  ) (_⊑_ : P → P → 𝓣 ̇  )
+       where
+
+ open PosetAxioms
+
+ Definition-3-1 : 𝓤 ⊔ 𝓣 ̇
+ Definition-3-1 = is-prop-valued _⊑_
+                × is-reflexive _⊑_
+                × is-transitive _⊑_
+
+ Definition-3-2 : 𝓤 ⊔ 𝓣 ̇
+ Definition-3-2 = Definition-3-1 × is-antisymmetric _⊑_
+
+ Lemma-3-3 : is-prop-valued _⊑_
+           → is-reflexive _⊑_
+           → is-antisymmetric _⊑_
+           → is-set P
+ Lemma-3-3 = posets-are-sets _⊑_
+
+ module _ (𝓥 : Universe) where
+  open import DomainTheory.Basics.Dcpo pt fe 𝓥
+
+  Definition-3-4 : {I : 𝓥 ̇  } → (I → P) → (𝓥 ⊔ 𝓣 ̇  ) × (𝓥 ⊔ 𝓣 ̇  )
+  Definition-3-4 {I} α = is-semidirected _⊑_ α , is-directed _⊑_ α
+
+  Remark-3-5 : {I : 𝓥 ̇  } (α : I → P)
+             → is-directed _⊑_ α
+             ＝ ∥ I ∥ × ((i j : I) → ∥ Σ k ꞉ I , (α i ⊑ α k) × (α j ⊑ α k) ∥)
+  Remark-3-5 α = refl
+
+  Definition-3-6 : {I : 𝓥 ̇  } → P → (I → P) → (𝓥 ⊔ 𝓣 ̇  ) × (𝓤 ⊔ 𝓥 ⊔ 𝓣 ̇  )
+  Definition-3-6 {I} x α = (is-upperbound _⊑_ x α) , is-sup _⊑_ x α
+
+  Definition-3-6-ad : poset-axioms _⊑_
+                    → {I : 𝓥 ̇  } (α : I → P)
+                    → {x y : P} → is-sup _⊑_ x α → is-sup _⊑_ y α → x ＝ y
+  Definition-3-6-ad pa {I} α = sups-are-unique _⊑_ pa α
+
+  Definition-3-7 : 𝓥 ⁺ ⊔ 𝓤 ⊔ 𝓣 ̇
+  Definition-3-7 = is-directed-complete _⊑_
+
+  Definition-3-7-ad : (𝓓 : DCPO {𝓤} {𝓣}) {I : 𝓥 ̇}
+                      {α : I → ⟨ 𝓓 ⟩} → is-Directed 𝓓 α → ⟨ 𝓓 ⟩
+  Definition-3-7-ad = ∐
+
+  Remark-3-8 : poset-axioms _⊑_
+             → {I : 𝓥 ̇  } (α : I → P)
+             → is-prop (has-sup _⊑_ α)
+  Remark-3-8 = having-sup-is-prop _⊑_
+
+module _ (𝓥 : Universe) where
+ open import DomainTheory.Basics.Dcpo pt fe 𝓥
+
+ Definition-3-9 : {𝓤 𝓣 : Universe} → (𝓤 ⊔ 𝓥 ⊔ 𝓣) ⁺ ̇
+ Definition-3-9 {𝓤} {𝓣} = DCPO {𝓤} {𝓣}
+
+ -- Remark-3-10: No formalisable content.
```

Comment:
Is this a meta-mathematical remark?

Reply:
Yes, Remark 3.10 just says that it's important to keep track of universe levels.

Reply:
Maybe add this sentence in the comment, then?